[openimageio] Requires explicit find_package for OpenEXR

**Describe the bug**
The usage hints after installation of OpenImageIO say
```
    find_package(OpenImageIO CONFIG REQUIRED)
    target_link_libraries(main PRIVATE OpenImageIO::OpenImageIO)
```
However, that does not work on x64-linux (cmake 3.18.4):
```
CMake Error at .../scripts/buildsystems/vcpkg.cmake:582 (_add_executable):
  Target "main" links to target "OpenEXR::IlmThreadConfig" but the target was
  not found.  Perhaps a find_package() call is missing for an IMPORTED
  target, or an ALIAS target is missing?
Call Stack (most recent call first):
  CMakeLists.txt:10 (add_executable)
```
(Repeated several times for different OpenEXR targets.)
And with cmake 3.23.3:
```
CMake Error at .../installed/x64-linux/share/openimageio/OpenImageIOTargets.cmake:71 (set_target_properties):
  The link interface of target "OpenImageIO::OpenImageIO" contains:

    OpenEXR::IlmThreadConfig

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

**Environment**
- OS: x64-linux
- Compiler: GCC 10.2.1-6
- cmake: 3.18.4 and 3.23.3
- vcpkg: 3640e7cb (Jan 9), due to issue #29156

**To Reproduce**
Steps to reproduce the behavior:
1. ./vcpkg install openimageio
2. Create subdirectory `test/oiio_without_openexr` with the following files:
CMake_Lists.txt:
```
cmake_minimum_required(VERSION 3.18)
set(CMAKE_TOOLCHAIN_FILE ${CMAKE_SOURCE_DIR}/../../scripts/buildsystems/vcpkg.cmake)
project(oiio_without_openexr)
find_package(OpenImageIO CONFIG REQUIRED)
#find_package(OpenEXR)
add_executable(main main.cpp)
target_link_libraries(main PRIVATE OpenImageIO::OpenImageIO)
```
main.cpp:
```
int main() { }
```
run.sh:
```
#!/bin/bash
rm -fr build
mkdir build
cd build
cmake ..
```

**Expected behavior**
The explicit additional find_package() call for OpenEXR should not be necessary.

--- a/ports/openimageio/fix-openexr-target-missing.patch
+++ b/ports/openimageio/fix-openexr-target-missing.patch
@@ -1,0 +1,13 @@
+diff --git a/src/cmake/Config.cmake.in b/src/cmake/Config.cmake.in
+index 08401a2..4f2694e 100644
+--- a/src/cmake/Config.cmake.in
++++ b/src/cmake/Config.cmake.in
+@@ -24,6 +24,8 @@ if (NOT @OPENIMAGEIO_CONFIG_DO_NOT_FIND_IMATH@ AND NOT OPENIMAGEIO_CONFIG_DO_NOT
+     if (@OpenEXR_VERSION@ VERSION_GREATER_EQUAL 3.0)
+         find_dependency(Imath @Imath_VERSION@
+                         HINTS @Imath_DIR@)
++        find_dependency(OpenEXR @OpenEXR_VERSION@
++                        HINTS @OpenEXR_DIR@)
+     elseif (@OpenEXR_VERSION@ VERSION_GREATER_EQUAL 2.4 AND @FOUND_OPENEXR_WITH_CONFIG@)
+         find_dependency(IlmBase @OpenEXR_VERSION@
+                         HINTS @IlmBase_DIR@ @OpenEXR_DIR@)

--- a/ports/openimageio/portfile.cmake
+++ b/ports/openimageio/portfile.cmake
@@ -1,9 +1,7 @@
-vcpkg_minimum_required(VERSION 2022-10-12)
-
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO OpenImageIO/oiio
-    REF v${VERSION}
+    REF "v${VERSION}"
     SHA512 c7a4283b78197c262d8da31460ce8b07b44546f822142e32e6c1ea22376e1c4b9cfe9c39cc0994987c6c4f653c1f2764057944da97a3a090bf1bcb74a2a0b2c2
     HEAD_REF master
     PATCHES
@@ -15,6 +13,7 @@ vcpkg_from_github(
         fix-vs2019-encoding-conversion.patch
         qt6.patch
         more_qt6.patch
+        fix-openexr-target-missing.patch
 )
 
 file(REMOVE_RECURSE "${SOURCE_PATH}/ext")
@@ -93,7 +92,7 @@ file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/share/doc"
 vcpkg_fixup_pkgconfig()
 
 file(INSTALL "${CMAKE_CURRENT_LIST_DIR}/usage" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}")
-file(INSTALL "${SOURCE_PATH}/LICENSE.md" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE.md")
 file(READ "${SOURCE_PATH}/THIRD-PARTY.md" third_party)
 string(REGEX REPLACE
     "^.*The remainder of this file"

--- a/ports/openimageio/vcpkg.json
+++ b/ports/openimageio/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "openimageio",
   "version": "2.4.5.0",
-  "port-version": 4,
+  "port-version": 5,
   "description": "A library for reading and writing images, and a bunch of related classes, utilities, and application.",
   "homepage": "https://github.com/OpenImageIO/oiio",
   "license": "BSD-3-Clause",

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -5694,7 +5694,7 @@
     },
     "openimageio": {
       "baseline": "2.4.5.0",
-      "port-version": 4
+      "port-version": 5
     },
     "openjpeg": {
       "baseline": "2.5.0",

--- a/versions/o-/openimageio.json
+++ b/versions/o-/openimageio.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "18125d8c18cd384804b12c5839331a6312c3ba6f",
+      "version": "2.4.5.0",
+      "port-version": 5
+    },
+    {
       "git-tree": "3c8df049a8e7b1c3e48006e78c4f5a9d451860a0",
       "version": "2.4.5.0",
       "port-version": 4

